contracts/Everlong.sol

```diff
@@ -215,36 +218,52 @@ contract Everlong is IEverlong {
     ///      maturity of the portfolio.
     /// @return Total amount of assets controlled by Everlong.
     function totalAssets() public view override returns (uint256) {
-        // If everlong holds no bonds, return the balance.
-        uint256 balance = ERC20(_asset).balanceOf(address(this));
-        if (_portfolio.totalBonds == 0) {
-            return balance;
-        }
+        return _totalAssets;
+    }
 
-        // Estimate the value of everlong-controlled positions by calculating
-        // the proceeds one would receive from closing a position with the
-        // portfolio's total amount of bonds and weighted average maturity.
-        // The weighted average maturity is rounded to the next checkpoint
-        // timestamp to underestimate the value.
-        return
-            balance +
-            IHyperdrive(hyperdrive).previewCloseLong(
-                asBase,
-                IEverlong.Position({
-                    maturityTime: IHyperdrive(hyperdrive)
-                        .getCheckpointIdUp(_portfolio.avgMaturityTime)
-                        .toUint128(),
-                    bondAmount: _portfolio.totalBonds
-                }),
-                ""
-            );
+    /// @notice Returns an approximate lower bound on the amount of assets
+    ///         received from redeeming the specified amount of shares.
+    /// @param _shares Amount of shares to redeem.
+    /// @return assets Amount of assets that will be received.
+    function previewRedeem(
+        uint256 _shares
+    ) public view override returns (uint256 assets) {
+        // Convert the share amount to assets.
+        assets = convertToAssets(_shares);
+
+        // Apply losses incurred by the portfolio.
+        uint256 losses = _calculatePortfolioLosses().mulDivDown(
+            assets,
+            _totalAssets
+        );
+
+        // If the losses from closing immature positions exceeds the assets
+        // owed to the redeemer, set the assets owed to zero.
+        if (losses > assets) {
+            // NOTE: We return zero since `previewRedeem` must not revert.
+            assets = 0;
+        }
+        // Decrement the assets owed to the redeemer by the amount of losses
+        // incurred from closing immature positions.
+        else {
+            assets -= losses;
+        }
     }
 
+    // TODO: Do not rebalance on deposit. This change will require updating
```

Comment:
We're also not going to update `_totalAssets` every deposit, right? Or at least, we'll just increase the `_totalAssets` by the deposit amount?

Reply:
Really great callout, I think we should just increment `_totalAssets` by the deposit amount.